### PR TITLE
ci: add dev mode to vagrant tests

### PR DESCRIPTION
**Pull request by lebauce**
skip skydive-cdd-overview-tests skip skydive-coverage skip skydive-devstack-tests skip skydive-helm-tests skip skydive-kolla-tests skip skydive-skydive-release skip skydive-create-binaries skip skydive-compile-tests skip skydive-functional-tests-backend-elasticsearch skip skydive-functional-tests-backend-orientdb skip skydive-go-fmt skip skydive-k8s-tests skip skydive-packaging-tests skip skydive-python-tests skip skydive-scale-tests skip skydive-selenium-tests skip skydive-unit-tests

### CODE REVIEW

**Comment by lebauce** [comment]
run vagrant-tests

**Comment by lebauce** [comment]
run vagrant-tests

**Comment by lebauce** [comment]
run vagrant-tests

**Comment by lebauce** [comment]
run vagrant-tests

**Comment by lebauce** [comment]
run vagrant-tests

File: scripts/ci/run-vagrant-tests.sh [hunk 1]
```diff
@@ -23,8 +23,20 @@ trap vagrant_cleanup EXIT
 export ANALYZER_COUNT=2
 export AGENT_COUNT=1
 
-for mode in dev binary package container
+for mode in binary package container
 do
+  if [ "$DEVMODE" == "true" ]; then
+    case $mode in
+      dev)
+        make install ;;
```
Comment (nplanel) [inline scripts/ci/run-vagrant-tests.sh]: need make -C  ${GOPATH}/src/github.com/skydive-project/skydive for all make call

File: scripts/ci/run-vagrant-tests.sh [hunk 2]
```diff
@@ -23,8 +23,20 @@ trap vagrant_cleanup EXIT
...
+      container)
+        make docker-image
+        docker save skydive/skydive:devel -o skydive-docker-devel.tar ;;
+      package)
+        make rpm ;;
```
Comment (nplanel) [inline scripts/ci/run-vagrant-tests.sh]: add  BOOTSTRAP_ARGS=-l  as you expect to find skydive/rpmbuild/RPMS/...

File: contrib/ansible/roles/skydive_analyzer/tasks/docker.yml [hunk 3]
```diff
@@ -10,6 +10,34 @@
     state: started
     enabled: yes
 
+- name: Run docker registry
+  shell: "docker run -d -p 5000:5000 --restart=always --name registry registry:2"
+  when: '"file://" in skydive_binary_remote_location'
```
Comment (safchain) [inline contrib/ansible/roles/skydive_analyzer/tasks/docker.yml]: not sure you need to surround the while line by with '

File: contrib/ansible/roles/skydive_analyzer/tasks/docker.yml [hunk 4]
```diff
@@ -10,6 +10,34 @@
...
+  set_fact: "skydive_docker_registry=localhost:5000"
+  when: '"file://" in skydive_binary_remote_location'
+
+- name: Copy docker image
+  copy:
+    src: "{{ skydive_binary_remote_location | replace('file://', '') }}"
+    dest: /tmp
+    force: true
+    mode: 0755
+  when: '"file://" in skydive_binary_remote_location'
```
Comment (safchain) [inline contrib/ansible/roles/skydive_analyzer/tasks/docker.yml]: same here and below

File: scripts/ci/run-vagrant-tests.sh [hunk 5]
```diff
@@ -23,8 +23,20 @@ trap vagrant_cleanup EXIT
 export ANALYZER_COUNT=2
 export AGENT_COUNT=1
 
-for mode in dev binary package container
+for mode in binary package container
 do
+  if [ "$DEVMODE" == "true" ]; then
+    case $mode in
+      dev)
```
Comment (safchain) [inline scripts/ci/run-vagrant-tests.sh]: dev is not in the list, I guess it should be binary

File: scripts/ci/run-vagrant-tests.sh (hunk 1 above)
Comment (safchain) [inline scripts/ci/run-vagrant-tests.sh]: make static ?, already done line 14

File: contrib/ansible/roles/skydive_analyzer/tasks/docker.yml (hunk 3 above)
Comment (lebauce) [reply contrib/ansible/roles/skydive_analyzer/tasks/docker.yml]: it seems so

File: scripts/ci/run-vagrant-tests.sh (hunk 1 above)
Comment (lebauce) [reply scripts/ci/run-vagrant-tests.sh]: fixed

File: scripts/ci/run-vagrant-tests.sh (hunk 5 above)
Comment (lebauce) [reply scripts/ci/run-vagrant-tests.sh]: fixed

**Comment by lebauce** [comment]
run vagrant-tests

File: scripts/ci/run-vagrant-tests.sh [hunk 6]
```diff
@@ -11,7 +11,13 @@ set -v
 dir="$(dirname "$0")"
 
 cd ${GOPATH}/src/github.com/skydive-project/skydive
-make static
+
+if [ "$DEVMODE" == "true" ]; then
+    make static
+    make rpm
```
Comment (nplanel) [inline scripts/ci/run-vagrant-tests.sh]: BOOTSTRAP_ARGS=-l

Comment (lebauce) [reply scripts/ci/run-vagrant-tests.sh]: fixed

**Comment by lebauce** [comment]
run vagrant-tests

**Pull request merged**